From 31e6be7438278dfa6b6a454e8a88b124d30c7b6f Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 24 Jan 2021 09:31:20 +0200
Subject: [PATCH] [SW-30548] uart: moving keypress print to run function

since uart work time is fairly long,  moving this print to the run
function which is executed in 'slow' context

Change-Id: Ic50f041bbb4a058234fc02744c08b84a23a3c6f3
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/module_uart.c b/preboot/module_uart.c
index be24344..6062d18 100644
--- a/preboot/module_uart.c
+++ b/preboot/module_uart.c
@@ -14,8 +14,6 @@
 	if (!preboot_uart_tstc())
 		return NOT_FOUND;
 
-	preboot_puts("Keyboard press received\n");
-
 	return FOUND_SLOW_RETURN;
 }
 
@@ -67,5 +65,6 @@
 	if (val < 0)
 		return;
 
+	preboot_puts("Keyboard press received\n");
 	hl_parse_uart(val);
 }
